app/apollo/resolvers/channel.js

```diff
@@ -414,29 +409,45 @@ const channelResolvers = {
           throw new RazeeValidationError(context.req.t('{{serSubCount}} service subscriptions depend on this channel version. Please have them updated/removed before removing this channel version.', {'serSubCount':serSubCount}), context);
         }
 
-        const channel_uuid = deployableVersionObj.channel_id;
-        const channel = await models.Channel.findOne({ uuid: channel_uuid, org_id });
+        // Get the Channel for the Version
+        const channel = await models.Channel.findOne({ versions: { $elemMatch: { uuid: uuid } }, org_id });
         if(!channel){
-          throw new NotFoundError(context.req.t('channel uuid "{{channel_uuid}}" not found', {'channel_uuid':channel_uuid}), context);
+          // If unable to find the Channel then cannot verify authorization, so throw an error
+          throw new NotFoundError(context.req.t('channel for version "{{uuid}}" not found', {'uuid':uuid}), context);
```

Comment:
If a version exists but the channel does not have a reference to it, the version will never be deleted?

Reply:
Correct -- authorization is checked on the Channel, so there's no way to delete a Version that is not associated with a channel.

Reply:
I mean if a version is associated with a channel, but not in the channel's versions array, then there will be no way to delete it.

Reply:
There is no known scenario where the Version will still exist but the Channel will no longer reference it, as the Channel reference is removed _after_ the Version is successfully deleted.  However, I've added a commit to handle this scenario as well, in case of future scenarios.